server/src/main/java/org/elasticsearch/health/node/selection/HealthNodeSelectorTaskParams.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.node.selection;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.persistent.PersistentTaskParams;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.elasticsearch.health.node.selection.HealthNodeSelector.TASK_NAME;
+
+class HealthNodeSelectorTaskParams implements PersistentTaskParams {
```

Comment:
Since this class doesn't have any actual data. I think we can do something with
```java
private static final HealthNodeSelectorTaskParams INSTANCE = new HealthNodeSelectorTaskParams();
```

and then we can make the constructor private and return the singleton instance in places like `fromXContent` and not have to have a parser at all.

Reply:
I will give it a go.

Reply:
If I do this, I still need to tell the parser how to skip that object right? If I just do this:

```
public static HealthNodeSelectorTaskParams fromXContent(XContentParser parser) {
        return INSTANCE;
    }
```
Then the parser would still be pointing to the beginning of the object, which I am assuming can create problems. With a quick scan of the API I didn't find a way to skip the value without introducing a parser. Did I miss anything?

If this is right and we still need to introduce a parser, I am not so sure if this is worth it... It will decrease object creation but I do not think this is a worthy optimization in this case. WDYT?

Reply:
I could keep the constructor private and expose the `INSTANCE` to avoid creating extra instances when creating the task :).

Reply:
If we do this then the tests for this class won't work anymore because they are expecting different objects to be created. I would prefer to leave it like this so it's inline with the other similar objects. What do you think?

Reply:
I think the test can override `assertEqualInstances` to make it match our new possible invariant? (ie. same instance?)

Reply:
The only way to go around it, that I could think of is this:

```
    public static final ObjectParser<HealthNodeTaskParams, Void> PARSER = new ObjectParser<>(
        TASK_NAME,
        true,
        () -> INSTANCE
    );
```

Reply:
That looks alright to me. @dakrone you'd be alright with this?